components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go

```diff
@@ -94,24 +94,55 @@ func (spec *FunctionSpec) defaultReplicas(config *DefaultingConfig, fn *Function
 }
 
 func (spec *FunctionSpec) defaultFunctionResources(config *DefaultingConfig, fn *Function) {
-	resources := spec.ResourceConfiguration.Function.Resources
+	var resources *corev1.ResourceRequirements
+	if spec.ResourceConfiguration != nil && spec.ResourceConfiguration.Function != nil && spec.ResourceConfiguration.Function.Resources != nil {
+		resources = spec.ResourceConfiguration.Function.Resources
+	}
 	defaultingConfig := config.Function.Resources
 	resourcesPreset := mergeResourcesPreset(fn, FunctionResourcesPresetLabel, defaultingConfig.Presets, defaultingConfig.DefaultPreset, defaultingConfig.RuntimePresets)
+	calculatedResources := defaultResources(resources, resourcesPreset.RequestMemory, resourcesPreset.RequestCPU, resourcesPreset.LimitMemory, resourcesPreset.LimitCPU)
+
+	if spec.ResourceConfiguration == nil {
+		spec.ResourceConfiguration = &ResourceConfiguration{}
+	}
+
+	if spec.ResourceConfiguration.Function == nil {
+		spec.ResourceConfiguration.Function = &ResourceRequirements{}
+	}
 
-	spec.ResourceConfiguration.Function.Resources = defaultResources(&resources, resourcesPreset.RequestMemory, resourcesPreset.RequestCPU, resourcesPreset.LimitMemory, resourcesPreset.LimitCPU)
+	spec.ResourceConfiguration.Function.Resources = calculatedResources
```

Comment:
We can extract it to separate function like `setFunctionResources`

Reply:
Done